pandas/core/indexes/base.py

```diff
@@ -6829,7 +6838,7 @@ def slice_locs(self, start=None, end=None, step=None) -> tuple[int, int]:
 
         return start_slice, end_slice
 
-    def delete(self, loc) -> Self:
+    def delete(self, loc: int | list[int]) -> Self:  # type: ignore[override]
```

Comment:
```suggestion
    def delete(self, loc: int | list[int]) -> Self:
```

looks like that is not needed